src/main/java/com/contrast/labs/ai/mcp/contrast/AssessService.java

```diff
@@ -217,89 +200,71 @@ public List<VulnLight> listVulnsByAppId(
-            return new ArrayList<>();
+            return returnVulns;
+        } catch (Exception e) {
+            logger.error("Error listing vulnerabilities for application: {}", appID, e);
+            throw new IOException("Failed to list vulnerabilities: " + e.getMessage(), e);
         }
     }
 
 
-    @Tool(name = "list_vulnerabilities_by_application_and_latest_session", description = "Takes an application name ( app_name ) and returns a list of vulnerabilities for the latest session matching that application name. This is useful for getting the most recent vulnerabilities without needing to specify session metadata.")
-    public List<VulnLight> listVulnsInAppByNameForLatestSession(
-            @ToolParam(description = "Application name") String app_name) throws IOException {
-        logger.info("Listing vulnerabilities for application: {}", app_name);
+    @Tool(name = "list_vulns_by_app_latest_session", description = "Takes an application ID (appID) and returns a list of vulnerabilities for the latest session matching that application ID. This is useful for getting the most recent vulnerabilities without needing to specify session metadata. Use list_applications_with_name first to get the application ID from a name.")
+    public List<VulnLight> listVulnsByAppIdForLatestSession(
+            @ToolParam(description = "Application ID") String appID) throws IOException {
+        logger.info("Listing vulnerabilities for application: {}", appID);
         ContrastSDK contrastSDK = SDKHelper.getSDK(hostName, apiKey, serviceKey, userName,httpProxyHost, httpProxyPort);
 
+        try {
+            SDKExtension extension = new SDKExtension(contrastSDK);
+            SessionMetadataResponse latest = extension.getLatestSessionMetadata(orgID, appID);
 
-        logger.debug("Searching for application ID matching name: {}", app_name);
-        Optional<Application> application = SDKHelper.getApplicationByName(app_name, orgID, contrastSDK);
-
-        if(application.isPresent()) {
-            try {
-                SDKExtension extension = new SDKExtension(contrastSDK);
-                SessionMetadataResponse latest = extension.getLatestSessionMetadata(orgID,application.get().getAppId());
-
-                // Use SDK's native TraceFilterBody with agentSessionId field
-                com.contrastsecurity.models.TraceFilterBody filterBody = new com.contrastsecurity.models.TraceFilterBody();
-                if(latest!=null&&latest.getAgentSession()!=null&&latest.getAgentSession().getAgentSessionId()!=null) {
-                    filterBody.setAgentSessionId(latest.getAgentSession().getAgentSessionId());
-                }
+            // Use SDK's native TraceFilterBody with agentSessionId field
+            com.contrastsecurity.models.TraceFilterBody filterBody = new com.contrastsecurity.models.TraceFilterBody();
+            if(latest!=null&&latest.getAgentSession()!=null&&latest.getAgentSession().getAgentSessionId()!=null) {
```

Comment:
the lack of spacing between operators makes this pretty hard to read.  Also, it would read better if each boolean expression is put on its own line.